lib/travis/remote_vcs/user.rb

```diff
@@ -25,7 +25,7 @@ def authenticate(provider: :github, code:, redirect_uri:)
         end
         return JSON.parse(resp.body)['data'] if resp.success?
 
-        halt 401, 'unauthorized'
+        raise ConnectionError
```

Comment:
@weban, looks like the name of the error doesn't reflect what actually happens. We got a response from VCS service, which means that it's not a "connection" error, but the "response" is not what we expect